modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/ContainerServicesMuleContextConfigurator.java

```diff
@@ -34,7 +35,13 @@ public ContainerServicesMuleContextConfigurator(ServiceRepository serviceReposit
 
   @Override
   public void configure(CustomizationService customizationService) {
-    serviceRepository.getServices()
-        .forEach(service -> customizationService.registerCustomServiceImpl(service.getName(), service));
+    serviceRepository.getServices().forEach(service -> {
+      String name = service.getName();
+      String contract = service.getContractName();
```

Comment:
When is the contract name anything but the default?

Reply:
look at LazyServiceProxy and the Service Javadoc